language/polkavm/move-to-polka/tests/examples_basic_test.rs

```diff
@@ -75,6 +75,80 @@ pub fn test_string() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+#[serial]
+pub fn test_string_index_of() -> anyhow::Result<()> {
+    initialize_logger();
+    let (mut instance, mut allocator) = new_move_program(
+        "output/string.polkavm",
+        "../examples/basic/sources/string.move",
+        vec![],
+    )?;
+    let result = instance
+        .call_typed_and_get_result::<u64, ()>(&mut allocator, "index_of", ())
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+    assert_eq!(result, 7);
+
+    Ok(())
+}
+
+#[test]
+#[serial]
+pub fn test_string_substring() -> anyhow::Result<()> {
+    initialize_logger();
+    let (mut instance, mut allocator) = new_move_program(
+        "output/string.polkavm",
+        "../examples/basic/sources/string.move",
+        vec![],
+    )?;
+    instance
+        .call_typed_and_get_result::<(), ()>(&mut allocator, "substring", ())
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    Ok(())
```

Comment:
Nit: It seems like we are only debugging the substring. Maybe we could improve this test to have a "hard" assert.